src/fides/api/util/aws_util.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Optional
+
+from boto3 import Session
+from botocore.exceptions import ClientError
+from loguru import logger
+
+from fides.api.common_exceptions import StorageUploadError
+from fides.api.schemas.storage.storage import AWSAuthMethod, StorageSecrets
+
+
+def get_aws_session(
+    auth_method: str,
+    storage_secrets: Dict[StorageSecrets, Any],
+    assume_role_arn: Optional[str] = None,
+) -> Session:
+    """
+    Abstraction to retrieve an AWS Session using secrets.
+
+    If an `assume_role_arn` is provided, the secrets will be used to
+    assume that role and return a Session instantiated with that role.
+    """
+    sts_client = None
+    if auth_method == AWSAuthMethod.SECRET_KEYS.value:
+        if storage_secrets is None:
+            err_msg = "Storage secrets not found for S3 storage."
+            logger.warning(err_msg)
+            raise StorageUploadError(err_msg)
+
+        session = Session(
+            aws_access_key_id=storage_secrets[StorageSecrets.AWS_ACCESS_KEY_ID.value],  # type: ignore
+            aws_secret_access_key=storage_secrets[
+                StorageSecrets.AWS_SECRET_ACCESS_KEY.value  # type: ignore
+            ],
+        )
+        # Check that credentials are valid
+        sts_client = session.client("sts")
+        sts_client.get_caller_identity()
+    elif auth_method == AWSAuthMethod.AUTOMATIC.value:
+        session = Session()
+        logger.info("Successfully created automatic session")
+    else:
+        logger.error("Auth method not supported for S3: {}", auth_method)
+        raise ValueError(f"Auth method not supported for S3: {auth_method}")
+
+    if assume_role_arn:
+        if sts_client is None:
```

Comment:
i wasn't sure if it was safe to call the 
```
sts_client = session.client("sts")
sts_client.get_caller_identity()
```
block for all use cases with the "automatic"  auth method, since we hadn't been doing that before. so rather than risk a regression, i'm only creating the sts client when secret keys are provided (as it had been done before), OR unless it's explicitly needed to assume role...the code is a bit harder to follow/less DRY, but i'd rather be safe for now...